### PR TITLE
Extend typings and tests

### PR DESCRIPTION
**Pull request by przemyslawzalewski**
Modules to be covered:

- [X] array
- [X] async
- [ ] date - @kendyl93 works on that in https://github.com/sandstreamdev/std/pull/31
- [x] debug
- [X] encoding
- [x] file
- [x] function
- [ ] is
- [X] math
- [ ] object
- [x] query
- [ ] range
- [x] regex
- [x] string
- [x] vector2
- [x] web

Let's cover other modules in a separate PR.

Current coverage status:
chrome_2019-12-13_13-46-26

### CODE REVIEW

File: vector2/add.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "add",
-  "description": "TODO: Fill short description here.",
+  "description": "Adds two vectors together.",
```
Comment (marekrozmus) [inline vector2/add.json]: do we need "together"?

File: vector2/multiply.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "multiply",
-  "description": "TODO: Fill short description here.",
+  "description": "Multiples two matrices together.",
```
Comment (marekrozmus) [inline vector2/multiply.json]: same here - isn't just "Multiple two matrices" enough?

File: README.md [hunk 3]
```diff
@@ -50,90 +50,164 @@ Checks if given arguments are all `Arrays`.
...
 
+Generates an array of numbers from 0 to n - 1.
+
 #### repeat
 
+Repeats the given element by given count of times.
+
 #### reverse
 
+Reverses the given array without mutating it (in contrast to Array.sort).
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply README.md]: Was meant to be `reverse` 😉

File: encoding/base64url.test.ts [hunk 4]
```diff
@@ -1,9 +1,118 @@
...
+    expect(decode(encode(unicodeText))).toEqual(unicodeText);
+  });
+
+  it("uses UTF-8 properly", () => {
+    expect(fromByteString(toByteString(range(256)))).toEqual(range(256));
+  });
+
+  it("converts base64 to base64URL", () => {
+    expect(fromByteString(toByteString(range(256)))).toEqual(range(256));
+  });
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply encoding/base64url.test.ts]: This is some leftover. Gonna remove that.

File: file/validName.test.ts [hunk 5]
```diff
@@ -2,8 +2,54 @@
...
+    expect(validName("test\0sas")).toBe(false);
+  });
+
+  it("allows proper file names", () => {
+    expect(validName("image.png")).toBe(true);
+    expect(validName("DATA.bin")).toBe(true);
+    expect(validName(".hidden")).toBe(true);
+    expect(validName("file with spaces")).toBe(true);
+    expect(validName("dashes-allowed")).toBe(true);
+    expect(validName("Mixed case")).toBe(true);
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply file/validName.test.ts]: Sure thing.

File: vector2/add.json (hunk 1 above)
Comment (przemyslawzalewski) [reply vector2/add.json]: I will remove that.

File: vector2/multiply.json (hunk 2 above)
Comment (przemyslawzalewski) [reply vector2/multiply.json]: Agreed.

File: array/differs.test.ts [hunk 6]
```diff
@@ -3,7 +3,18 @@
...
+    expect(differs([2, 3], [1, 2, 5])).toBe(true);
+  });
+
+  it("short-circuits over parameter presence", () => {
+    expect(differs(null, [1, 2])).toBe(true);
+    expect(differs([2, 3], undefined)).toBe(true);
+  });
+
+  it("compares elements index-wise", () => {
+    expect(differs([1, 2, 3], [1, 2, 3])).toBe(false);
```
[the remark this answers is not in the document]
Comment (przemyslawzalewski) [reply array/differs.test.ts]: Thanks, I am pushing the new test case.

**Pull request merged**